src/plugins/elasticsearch/index.js

```diff
@@ -49,12 +51,30 @@ module.exports = function (kibana) {
         return reply.continue();
       }
 
+      function noCreateIndex(request, reply) {
+        const requestPath = _.trim(request.path);
+        const matchPath = createProxy.createPath(config.get('kibana.index'));
+
+        if (requestPath === matchPath) {
+          return reply({
+            error: 'You cannot modify the primary kibana index through this interface.'
+          }).code(405).takeover();
```

Comment:
It would be a bit cleaner to simply reply with a Boom error here e.g. `return reply(Boom.methodNotAllowed('You cannot modify the primary kibana index through this interface.'));`

The request would be taken over automatically and you'd get a response that looks like this:

```
{
    "statusCode": 405,
    "error": "Method Not Allowed",
    "message": "You cannot modify the primary kibana index through this interface."
}
```